shared/js/dataDriver.js

```diff
@@ -85,6 +85,18 @@ loop.DataDriver = function() {
     }
 
     /**
+      * Add a page to the room's table of contents.
+      *
+      * @param {String} url       The url of the page.
+      * @param {String} title     The title of the page.
+      * @param {Object} metadata  Additional metadata,
```

Comment:
Ok, so you want the method call to take a single parameter but you still want the firebase record to have the values in this object pulled up to the top level of the firebase record? (That's the extra level of nesting I was talking about) No wonder I was so confused.